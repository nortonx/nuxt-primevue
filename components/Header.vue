<template>
  <Menubar :model="items">
    <template #start>
      <svg width="35" height="40" viewBox="0 0 35 40" fill="none" xmlns="http://www.w3.org/2000/svg">
          <path d="..." fill="var(--p-text-color)" />
          <path d="..." fill="transparent" />
      </svg>
    </template>
    <template #item="{ item }">
      <NuxtLink :to="item.to">
        <i :class="item.icon"></i>
        <span class="ml-1 mr-5">{{ item.label }}</span>
      </NuxtLink>
    </template>
  </Menubar>
</template>

<script setup lang="ts">
import Menubar from "primevue/menubar";
import { computed } from "vue"

const items = computed(() => [
  {
    label: "Home",
    icon: "pi pi-fw pi-home",
    to: "/"
  },
  {
    label: "Word Counter",
    icon: "pi pi-fw pi-pencil",
    to: "/wordcounter"
  },
  {
    label: "ToDo List",
    icon: "pi pi-fw pi-list",
    to: "/todo"
  },
  {
    label: "Data",
    icon: "pi pi-fw pi-database",
    to: "/data"
  },
  {
    label: "Form",
    icon: "pi pi-fw pi-user",
    to: "/form"
  },
  {
    label: "Orders",
    icon: "pi pi-fw pi-shopping-cart",
    to: "/orders"
  },
  {
    label: "Parking",
    icon: "pi pi-fw pi-car",
    to: "/parking"
  },
  {
    label: "About",
    icon: "pi pi-fw pi-info",
    to: "/about"
  }
])
</script>

<style lang="scss" scoped>
.p-menubar {
  .p-menuitem {
    .p-menuitem-content {
      a {
        text-decoration: none;
      }
    }
  }
}
</style>