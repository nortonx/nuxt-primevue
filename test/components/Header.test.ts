/*
  A unit test with "mount" would be an integrated test since it mounts the component and its children.
  A unit test with "shallowMount" would be a unit test since it mounts the component only.
*/
import { shallowMount } from "@vue/test-utils";
import Header from "@/components/Header.vue";
import { describe, it, expect, beforeEach } from "vitest";

describe("Header component", () => {
  let wrapper: any;

  const items = [
    {
      label: "Home",
      icon: "home",
      to: "/",
    },
    {
      label: "About",
      icon: "info",
      to: "/about",
    },
  ];

  beforeEach(() => {
    wrapper = shallowMount(Header, { props: { items } });
  });
  it("it should render correctly and match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
