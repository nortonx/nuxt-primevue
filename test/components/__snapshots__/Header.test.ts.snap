// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > it should render correctly and match snapshot 1`] = `"<menubar-stub data-v-a81738bd="" model="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" items="[object Object],[object Object]"></menubar-stub>"`;
